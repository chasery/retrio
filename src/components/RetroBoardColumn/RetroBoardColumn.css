.RetroBoardColumn {
  padding: 1.5rem 1rem;
  overflow-y: scroll;
}

.RetroBoardColumn + .RetroBoardColumn {
  border-top: 0.0625rem solid #ebebeb;
}

.RetroBoardColumn__title {
  display: flex;
  align-items: center;
  margin-bottom: 1.5rem;
}
.RetroBoardColumn__title > span {
  flex-grow: 1;
}
.RetroBoardColumn__button {
  display: block;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 0.25rem;
  text-decoration: none;
  transition: background-color 0.25s ease;
}
.RetroBoardColumn__button:hover {
  box-shadow: inset 0 0.0625rem 0.0625rem #0000001c;
}
.RetroBoardColumn__button.primary {
  background-color: #2660a4;
  color: #fff;
}
.RetroBoardColumn__button.primary:hover {
  background-color: transparent;
  color: #2660a4;
}

@media all and (min-width: 30rem) {
}

@media all and (min-width: 50rem) {
  .RetroBoardColumn:nth-child(2) {
    border-top: none;
  }
  .RetroBoardColumn:nth-child(2) {
    border-top: none;
    border-left: 0.0625rem solid #ebebeb;
  }
  .RetroBoardColumn:nth-child(3) {
    border-top: 0.0625rem solid #ebebeb;
  }
  .RetroBoardColumn:nth-child(4) {
    border-left: 0.0625rem solid #ebebeb;
    border-top: 0.0625rem solid #ebebeb;
  }
}

@media all and (min-width: 64rem) {
  .RetroBoardColumn {
    overflow-y: visible;
  }
  .RetroBoardColumn + .RetroBoardColumn {
    border-left: 0.0625rem solid #ebebeb;
  }
  .RetroBoardColumn:nth-child(3) {
    border-top: none;
  }
  .RetroBoardColumn:nth-child(4) {
    border-top: none;
  }
}

.RetroBoardColumn.good {
  background: #fbfbfb url('../../assets/icons/good.svg') no-repeat center center;
  background-size: 33%;
}
.RetroBoardColumn.bad {
  background: #fbfbfb url('../../assets/icons/bad.svg') no-repeat center center;
  background-size: 33%;
}
.RetroBoardColumn.toTry {
  background: #fbfbfb url('../../assets/icons/to-try.svg') no-repeat center
    center;
  background-size: 33%;
}
.RetroBoardColumn.shoutOut {
  background: #fbfbfb url('../../assets/icons/shout-out.svg') no-repeat center
    center;
  background-size: 33%;
}
