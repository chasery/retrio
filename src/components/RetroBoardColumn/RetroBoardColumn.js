import React from 'react';
import { Link } from 'react-router-dom';
import RetroBoardCard from '../RetroBoardCard/RetroBoardCard';
import './RetroBoardColumn.css';

function RetroBoardColumn(props) {
  const { boardId, title, cards } = props;

  const renderCards = (cards) => {
    if (cards) {
      return cards.map((card) => (
        <RetroBoardCard
          key={card.id}
          id={card.id}
          headline={card.headline}
          text={card.text}
          submitted_by={card.submitted_by}
        />
      ));
    }
  };

  return (
    <div className='RetroBoardColumn'>
      <h3 className='RetroBoardColumn__title'>
        <span>{title}</span>
        <Link to={`/boards/${boardId}/add-card`}>Add Card</Link>
      </h3>
      <ol>{renderCards(cards)}</ol>
    </div>
  );
}

export default RetroBoardColumn;
