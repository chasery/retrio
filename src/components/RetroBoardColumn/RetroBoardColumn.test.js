import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RetroBoardColumn from './RetroBoardColumn';

describe('RetroBoardColumn component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<RetroBoardColumn />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
