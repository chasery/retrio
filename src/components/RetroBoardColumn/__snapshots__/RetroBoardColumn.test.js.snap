// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroBoardColumn component renders without error 1`] = `
<div
  className="RetroBoardColumn"
>
  <h3
    className="RetroBoardColumn__title"
  >
    <span />
    <Link
      to="/boards/undefined/add-card"
    >
      Add Card
    </Link>
  </h3>
  <ol />
</div>
`;
